### PR TITLE
Fixed various XR minitest errors/failures.

### PR DESCRIPTION
**Pull request by chburket**
These changes should only affect XR.  I ran the modified minitests against n9kv, as well, to make sure I didn't break anything.

### CODE REVIEW

File: lib/cisco_node_utils/cmd_ref/show_version.yaml [hunk 1]
```diff
@@ -65,7 +65,8 @@ system_image:
   N6k: *sys_img_isan
   N7k: *sys_img_isan
   ios_xr:
-    get_value: '/IOS XR.*Version (.*)$/'
```
Comment (chburket) [inline lib/cisco_node_utils/cmd_ref/show_version.yaml:68]: We decided this was a better string to describe "system image" than just the simple version.

File: tests/test_interface_private_vlan.rb [hunk 2]
```diff
@@ -55,6 +55,12 @@ def interface_cleanup_pvlan
   end
 
   def test_pvlan_mapping
+    if validate_property_excluded?('interface', 'feature_vlan')
```
Comment (chburket) [inline tests/test_interface_private_vlan.rb:58]: `Interface.new('vlan13')` attempts to execute unsupported yaml, so I added this bail-out

File: tests/test_node_ext.rb [hunk 3]
```diff
@@ -379,7 +379,7 @@ def test_get_boot
   end
 
   def test_get_system
-    if validate_property_excluded?('system', 'resources')
```
Comment (chburket) [inline tests/test_node_ext.rb:382]: The old code was checking the wrong property.

File: tests/test_nxapi.rb [hunk 4]
```diff
@@ -44,6 +44,7 @@ def teardown
   end
 
   def cleanup
+    return if Cisco::Environment.environment[:port]
```
Comment (chburket) [inline tests/test_nxapi.rb:47]: `startup`/`teardown` are called even if all tests are skipped

File: tests/test_snmpuser.rb [hunk 5]
```diff
@@ -33,6 +33,7 @@ def setup
   end
 
   def create_user(name, opts='')
+    opts = 'network-admin v1 IPv4 none' if opts == '' && platform == :ios_xr
```
Comment (chburket) [inline tests/test_snmpuser.rb:36]: can't create an snmp user on XR with just a name

File: tests/test_yang.rb [hunk 6]
```diff
@@ -230,9 +230,10 @@ def test_errors
       node.replace_yang('{"Cisco-IOS-XR-infra-rsi-cfg:aabbcc": "foo"}')
     end
 
-    # for some reason replace_yang does not have the same error checking
-    # that merge_yang does, so this just fails quietly
-    node.replace_yang('{"Cisco-IOS-XR-infra-rsi-cfg:vrfs": }')
+    # Request is not wellformed
```
Comment (chburket) [inline tests/test_yang.rb:233]: the latest image of XR fixed this, apparently

**Comment by chrisvanheuveln** [comment]
Superceded by #438

**Pull request closed without merging**